### PR TITLE
Authentication class

### PR DESCRIPTION
**Pull request by gu1ll0me**
Added new class to manage authentification with a file, a sqlite database and with kwallet.

### CODE REVIEW

**Comment by Ram-Z** [comment]
looks ok to me, but I don't know python!

@Inkane @george2 what do you think?

File: ccrauth.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    def _set_info(self, username, password):
+        """ set the username and password
+        """
+        self.username = username
+        self.password = password
+
+    def store_auth_info(self, username, password):
+        """ store authentication information
+            reimplemented in child classes
```
Comment (rshipp) [inline ccrauth.py]: should be unindented to match the rest of the docstrings

File: ccrauth.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        """
+        pass
```
Comment (rshipp) [inline ccrauth.py]: perhaps raise a NotImplemented exception here?

**Comment by Inkane** [comment]
The code looks fine, but I'd split the ccrauth.py file into multiple files, at least 2 (one with "interface" and one for the implementations), maybe even more (one for each implementation). Else the library has dependencies, which you probably only want in a client (e.g. file backend for the CLI backend, KWallet for the GUI).

**Comment by rshipp** [comment]
+1 for splitting up the auth classes into multiple files.

**Comment by gu1ll0me** [comment]
Thanks for your advices. Tell me if you think of other improvements.

**Pull request merged**